web/src/js/authentication/user.js

```diff
@@ -330,3 +335,24 @@ export const reloadUser = async () => {
   }
   await user.reload()
 }
+
+export const updateUserEmail = async newEmail => {
+  const continueUrl = constructUrl(
+    accountURL,
+    { verified: true },
+    { absolute: true }
+  )
+  return new Promise((resolve, reject) => {
+    const user = firebase.auth().currentUser
+    user
+      .verifyBeforeUpdateEmail(newEmail, {
+        url: continueUrl,
+      })
+      .then(function() {
+        resolve()
+      })
+      .catch(function(error) {
+        reject(error)
+      })
+  })
```

Comment:
stylistic info: in an async/await context this promise could be written as just
`const user = firebase.auth().currentUser`
`await user.verifyBeforeUpdateEmail(newEmail, {        url: continueUrl,      })`
if there is an error in 'verifyBeforeUpdateEmail', the updateUserEmail promise will be reject
actually since async wraps and returns a promise you could also write this block as 
`const user = firebase.auth().currentUser`
`return user.verifyBeforeUpdateEmail(newEmail, {        url: continueUrl,      })`